### PR TITLE
Restructure load balancer status metrics

### PR DESCRIPTION
**Pull request by walbertus**
Restructure status metric from the value of UP/DOWN to more detailed metrics to capture different statuses.
Change applied for load balancer related components:
 - Load balancer
 - Load balancer pool
 - Load balancer pool member

Signed-off-by: William Albertus Dembo <w.albertusd@gmail.com>

### CODE REVIEW

**Comment by walbertus** [comment]
Address #59

File: collector/load_balancer_collector_test.go [hunk 1]
```diff
@@ -104,120 +104,267 @@ func TestLoadBalancerCollector_GenerateLoadBalancerStatusMetrics(t *testing.T) {
 			},
 			expectedMetrics: []loadBalancerStatusMetric{
 				{
-					ID:     "fake-load-balancer-id-01",
-					Name:   "fake-load-balancer-name-01",
-					Status: 1.0,
+					ID:   "fake-load-balancer-id-01",
+					Name: "fake-load-balancer-name-01",
+					StatusDetail: map[string]float64{
```
Comment (girikuncoro) [inline collector/load_balancer_collector_test.go]: the maps seem to be repeated in several places, can this be var/const instead defined outside?

Comment (girikuncoro) [reply collector/load_balancer_collector_test.go]: update: just realized the values are different in each case, how about extracting to function instead?

Comment (girikuncoro) [reply collector/load_balancer_collector_test.go]: something like `buildStatusDetail("UP", 1.0)` which will return map with default 0.0 for all key-value pairs except `UP`

Comment (walbertus) [reply collector/load_balancer_collector_test.go]: The logic to construct status detail is the same everywhere, wdyt if constructStatusDetail detached from loadBalancerCollector? Then we can use this function on the test and other metric

Comment (girikuncoro) [reply collector/load_balancer_collector_test.go]: or can accept slice of maps as well only for non-zero status
```
type nonZeroStates map[string]float64
...
buildStatusDetails(nonZeroStates{"UP": 1.0})
```

Comment (walbertus) [reply collector/load_balancer_collector_test.go]: Ah, that also works.
Will implement this

**Review by girikuncoro — APPROVED** [review]
/lgtm

**Pull request merged**